packages/utils/src/signatureUtils.ts

```diff
@@ -129,6 +130,18 @@ export function parseSignatureWithoutPrefix(
   throw new Error(`Unable to parse signature (expected signer ${signer})`)
 }
 
+export function guessEIP712TypedDataSigner(typedData: EIP712TypedData, signature: string): string {
```

Comment:
Its not really a "guess", so to align with Ethereum terminology I would recommend calling this `recoverEIP712TypedDataSigner`